### PR TITLE
Add EdDSA algorithm support

### PR DESCRIPTION
**Pull request by lritter14**
Allows for providers to use EdDSA algorithm but does not support verifying access tokens.

### CODE REVIEW

File: oidc/oidc.go [hunk 1]
```diff
@@ -450,6 +452,8 @@ func (i *IDToken) VerifyAccessToken(accessToken string) error {
 		h = sha512.New384()
 	case RS512, ES512, PS512:
 		h = sha512.New()
+	case EdDSA:
+		return errUnsupportedAlgorithm
```
Comment (ericchiang) [inline oidc/oidc.go]: wouldn't this be sha512.New()?

**Review by ericchiang — COMMENTED** [review]
Out of curiosity, what providers support this?

Would it be possible to add some tests that actually exercise the signing and verification? E.g. take a look at jwks_test.go verify_test.go

If not, I can take a crack at it

File: oidc/oidc.go (hunk 1 above)
Comment (lritter14) [reply oidc/oidc.go]: I believe that verifying EdDSA key with ed25519 protocol is a bit more complicated than the pattern that exists here [short explanation](https://security.stackexchange.com/questions/233099/validating-an-ed25519-public-key#:~:text=Yes.,see%20if%20the%20equation%20checks.)

**Comment by lritter14** [comment]
> Out of curiosity, what providers support this?
> 
> Would it be possible to add some tests that actually exercise the signing and verification? E.g. take a look at jwks_test.go verify_test.go
> 
> If not, I can take a crack at it

Here is an example provider that supports EdDSA. I was not originally planning on adding verification functionality as it is more complicated for ed25519 protocol.

File: oidc/oidc.go (hunk 1 above)
Comment (ericchiang) [reply oidc/oidc.go]: This is only for the at_hash value, it's not doing full verification :)

**Comment by ericchiang** [comment]
Did you mean to hyperlink something in that comment?

I don't think I'd want to merge this if it only implements partial support. I can add verification functionality when I get a sec.

**Comment by lritter14** [comment]
> Did you mean to hyperlink something in that comment?
> 
> I don't think I'd want to merge this if it only implements partial support. I can add verification functionality when I get a sec.

**Comment by lritter14** [comment]
I added some additional tests and took your suggestion for using sha512.New() to verify the hash. I am not super knowledgeable in this space so I am not confident in that change.

P.S. closing the PR was a mistake

**Review by ericchiang — APPROVED** [review]
Awesome, thanks for sending the full change. Lgtm :)

**Comment by ericchiang** [comment]
If there are any issues with the at_hash support, we can always fix it up later. I'm not super worried about it.

**Pull request merged**